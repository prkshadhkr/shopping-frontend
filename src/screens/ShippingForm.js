import React, { useState } from 'react';


const ShippingForm = ({ order_id, save } ) => {
  const initialValue = {
		order_id,
    address: "",
    city: "",
    zip_code: ""
  };
  const [formData, setFormData] = useState(initialValue);
  const handleSubmit = e => {
    e.preventDefault();
    save(formData);
  };
  const handleChange = e => {
    const { name, value } = e.target;
    setFormData(d => ({
      ...d,
      [name]: value
    }));
  };

  return (
    <div>
      <form onSubmit={handleSubmit} >
          <div className="form-group row">
						<label htmlFor="for-address" className="col-sm-2">Address</label>
							<div className="col-sm-10">
								<input 
									id="for-address"
									name="address"
									value={formData.address}
									onChange={handleChange}
									className="form-control"/>
							</div>
					</div>
          <div className="form-group row">
						<label htmlFor="for-city" className="col-sm-2">City</label>
							<div className="col-sm-10">
								<input 
									id="for-city"
									name="city"
									value={formData.city}
									onChange={handleChange}
									className="form-control"/>
							</div>
					</div>
          <div className="form-group row">
						<label htmlFor="for-zip_code" className="col-sm-2">Zip code</label>
							<div className="col-sm-10">
								<input 
									id="for-zip_code"
									name="zip_code"
									value={formData.zip_code}
									onChange={handleChange}
									className="form-control"/>
							</div>
					</div>

          <button 
						type="submit"
						className="btn btn-outline-primary float-right"
					>
						Add Shipping
					</button>
        </form>
    </div>
  )
}

export default ShippingForm;