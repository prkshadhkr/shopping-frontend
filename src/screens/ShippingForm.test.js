import React from 'react';
import { render } from '@testing-library/react';
import ShippingForm from './ShippingForm';
import { MemoryRouter } from 'react-router';

it ('renders without crashing', function() {
  render(
    <MemoryRouter>
      <ShippingForm />
    </MemoryRouter>
  )
});

it ('matches snapshot', function() {
  const { asFragment } = render (
    <MemoryRouter>
      <ShippingForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
})