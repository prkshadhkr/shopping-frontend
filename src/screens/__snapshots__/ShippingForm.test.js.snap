// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <div
        class="form-group row"
      >
        <label
          class="col-sm-2"
          for="for-address"
        >
          Address
        </label>
        <div
          class="col-sm-10"
        >
          <input
            class="form-control"
            id="for-address"
            name="address"
            value=""
          />
        </div>
      </div>
      <div
        class="form-group row"
      >
        <label
          class="col-sm-2"
          for="for-city"
        >
          City
        </label>
        <div
          class="col-sm-10"
        >
          <input
            class="form-control"
            id="for-city"
            name="city"
            value=""
          />
        </div>
      </div>
      <div
        class="form-group row"
      >
        <label
          class="col-sm-2"
          for="for-zip_code"
        >
          Zip code
        </label>
        <div
          class="col-sm-10"
        >
          <input
            class="form-control"
            id="for-zip_code"
            name="zip_code"
            value=""
          />
        </div>
      </div>
      <button
        class="btn btn-outline-primary float-right"
        type="submit"
      >
        Add Shipping
      </button>
    </form>
  </div>
</DocumentFragment>
`;
